fdbclient/ThreadSafeTransaction.cpp

```diff
@@ -149,9 +149,9 @@ ThreadSafeTransaction::ThreadSafeTransaction(DatabaseContext* cx, ISingleThreadT
 	auto tr = this->tr = ISingleThreadTransaction::allocateOnForeignThread(type);
 	// No deferred error -- if the construction of the RYW transaction fails, we have no where to put it
 	onMainThreadVoid(
-	    [tr, type, cx]() {
+	    [tr, cx]() {
 		    cx->addref();
-		    ISingleThreadTransaction::create(tr, type, Database(cx));
+		    tr->setDatabase(Database(cx));
```

Comment:
Are there any thread-safety concerns with this?

Reply:
The only place where `tr` can be accessed before `setDatabase` is called is when `ThreadSafeTransaction::getCommittedVersion` is called, but this should be thread-safe (but fragile, as before).

Reply:
It's a bit hard for me to reason fully about whether the previous code is thread-safe (I suppose it probably is if you call it after you have gotten a commit result, but maybe not before). The one thing that's different here if I'm understanding correctly is that we are modifying the value one extra time in this `setDatabase` call. 

Previously it was initialized in the `ThreadSafeTransaction` constructor and updated during commit, and the update during commit is potentially a problem if read concurrently. Of course this would be a misuse according to the documentation of the API.

Now it looks like it is initialized in the `ThreadSafeTransaction` constructor, updated on the main thread when `setDatabase` gets called, and then updated on `commit`. Perhaps this doesn't make a huge difference if the outcome is that accessing the committed version prior to `commit` is still thread-unsafe, though it's possible we could attempt to improve the situation using atomics or something. At any rate, it sounds like this PR doesn't make things worse and this shouldn't be a blocker.